4337/test/e2e/ReferenceEntryPoint.spec.ts

```diff
@@ -44,8 +44,13 @@ describe('E2E - Reference EntryPoint', () => {
   const deployEntryPoint = async (deployer: Signer, relayer: Signer) => {
     const { abi, bytecode } = EntryPointArtifact
     const transaction = await deployer.sendTransaction({ data: bytecode })
-    const receipt = await transaction.wait()
-    return new ethers.Contract(receipt!.contractAddress!, abi, relayer)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const receipt = (await transaction.wait())!
```

Comment:
I chose to disable `no-non-null-assertion` lints for all `transaction.wait()!` instances (there are more below as well). The rationale is that `receipt` will only be `null` if called with `wait(0)`, but not if called with `wait(n)` where `n > 0` or `wait()` (which defaults to `n = 1`). I consider this a typing issue in `ethers`, although I'm not familiar enough with TypeScript to know if it is possible or how to encode dependant types this way.

Reply:
It's possible with [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) but it has to be done on the ethers' side

Reply:
Thanks so much for the link... I went down a rabbit hole with this and found a better solution. In fact, you can augment the `ethers` module with additional overloads for these methods 🤯.

Anyway, pushed a new version that does not require the `eslint-disable-next-line` and gets rid of the `ParseLog` type and casting from 👇.